apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-content.tsx

```diff
@@ -0,0 +1,113 @@
+import type { InvalidValue, LayersValue } from "@webstudio-is/css-data";
+import { parseBoxShadow } from "@webstudio-is/css-data";
+import {
+  Flex,
+  Label,
+  Text,
+  TextArea,
+  theme,
+  Tooltip,
+} from "@webstudio-is/design-system";
+import { InformationIcon } from "@webstudio-is/icons";
+import { useState } from "react";
+import type { RenderCategoryProps } from "../../style-sections";
+import { addBoxShadow } from "./utils";
+
+type IntermediateValue = {
+  type: "intermediate";
+  value: string;
+};
+
+type BoxShadowEditProps = Pick<RenderCategoryProps, "createBatchUpdate"> & {
+  index: number;
+  value: string;
+  onEditLayer: (index: number, layers: LayersValue) => void;
+};
+
+type BoxShadowContentProps = Pick<RenderCategoryProps, "createBatchUpdate">;
+
+export const BoxShadowContent = (
+  props: BoxShadowContentProps | BoxShadowEditProps
+) => {
+  const [intermediateValue, setIntermediateValue] = useState<
+    IntermediateValue | InvalidValue | undefined
+  >(
+    "onEditLayer" in props
+      ? {
+          type: "intermediate",
+          value: props.value,
+        }
+      : undefined
+  );
+
+  const handleChange = (value: string) => {
+    setIntermediateValue({
+      type: "intermediate",
+      value,
+    });
+  };
+
+  const handleComplete = () => {
+    if (intermediateValue === undefined) {
+      return;
+    }
+    const layers = parseBoxShadow(intermediateValue.value);
```

Comment:
I have feeling it doesn't work as expected, and don't add layers to existing

Reply:
I am confused what I am supposed to do about this comment, it does work so 🤷‍♂️